rclcpp/include/rclcpp/publisher_options.hpp

```diff
@@ -0,0 +1,207 @@
+class PublisherOptions
+{
+public:
+  /// Create PublisherOptions with default values, optionally specifying the allocator to use.
+  /**
+   * Default values for the node options:
+   *
+   *   - context = rclcpp::contexts::default_context::get_global_default_context()
+   *   - arguments = {}
+   *   - initial_parameters = {}
+   *   - use_global_arguments = true
+   *   - use_intra_process_comms = false
+   *   - start_parameter_services = true
+   *   - start_publisher_publisher = true
+   *   - publisher_qos_profile = rmw_qos_profile_default
+   *   - allocator = rcl_get_default_allocator()
+   *
+   * \param[in] allocator allocator to use in construction of PublisherOptions.
+   */
+  RCLCPP_PUBLIC
+  PublisherOptions() = default;
+
+  RCLCPP_PUBLIC
+  explicit PublisherOptions(const rmw_qos_profile_t & qos_profile)
+  : publisher_qos_profile_(qos_profile) {}
+
+  /// Destructor.
+  RCLCPP_PUBLIC
+  virtual
+  ~PublisherOptions() = default;
+
+  /// Copy constructor.
+  RCLCPP_PUBLIC
+  PublisherOptions(const PublisherOptions & other) = default;
+
+  /// Assignment operator.
+  RCLCPP_PUBLIC
+  PublisherOptions &
+  operator=(const PublisherOptions & other) = default;
+
+
+  /// Return a reference to the publisher_qos_profile QoS.
+  RCLCPP_PUBLIC
+  const rmw_qos_profile_t &
+  qos_profile() const
+  {
+    return publisher_qos_profile_;
+  }
+
+  RCLCPP_PUBLIC
+  rmw_qos_profile_t &
+  qos_profile()
+  {
+    return publisher_qos_profile_;
+  }
+
+  /// Set the publisher_qos_profile QoS, return this for parameter idiom.
+  /**
+   * The QoS settings to be used for the publisher
+   */
+  RCLCPP_PUBLIC
+  PublisherOptions &
+  qos_profile(const rmw_qos_profile_t & publisher_qos_profile)
+  {
+    publisher_qos_profile_ = publisher_qos_profile;
+    return *this;
+  }
+
+  RCLCPP_PUBLIC
+  size_t &
+  qos_history_depth()
+  {
+    return publisher_qos_profile_.depth;
+  }
+
+  RCLCPP_PUBLIC
+  PublisherOptions &
+  qos_history_depth(size_t depth)
+  {
+    publisher_qos_profile_.depth = depth;
+    return *this;
+  }
+
+
+  RCLCPP_PUBLIC
+  const QOSDeadlineEventCallbackType &
+  deadline_callback() const
+  {
+    return callbacks_.deadline_callback_;
+  }
+
+  RCLCPP_PUBLIC
+  PublisherOptions &
+  deadline_callback(const QOSDeadlineEventCallbackType & callback)
+  {
+    callbacks_.deadline_callback_ = callback;
+    return *this;
+  }
+
+  RCLCPP_PUBLIC
+  const QOSLivelinessEventCallbackType &
+  liveliness_callback() const
+  {
+    return callbacks_.liveliness_callback_;
+  }
+
+  RCLCPP_PUBLIC
+  PublisherOptions &
+  liveliness_callback(const QOSLivelinessEventCallbackType & callback)
+  {
+    callbacks_.liveliness_callback_ = callback;
+    return *this;
+  }
+
+  RCLCPP_PUBLIC
+  const QOSLifespanEventCallbackType &
+  lifespan_callback() const
+  {
+    return callbacks_.lifespan_callback_;
+  }
+
+  RCLCPP_PUBLIC
+  PublisherOptions &
+  lifespan_callback(const QOSLifespanEventCallbackType & callback)
```

Comment:
C++ function style comments

Reply:
@ross-desmond What did you mean by this?

Reply:
@mm318 check this out: https://google.github.io/styleguide/cppguide.html#Function_Comments